packages/compiler-sfc/src/parse.ts

```diff
@@ -95,11 +96,18 @@ export function parse(
     sourceRoot = '',
     pad = false,
     ignoreEmpty = true,
-    compiler = CompilerDOM
+    compiler = CompilerDOM,
+    defaultScriptLang
   }: SFCParseOptions = {}
 ): SFCParseResult {
   const sourceKey =
-    source + sourceMap + filename + sourceRoot + pad + compiler.parse
+    source +
+    sourceMap +
+    filename +
+    sourceRoot +
+    pad +
+    compiler.parse +
+    (defaultScriptLang || '')
```

Comment:
`lang` field is optional in `SFCBlock` so we can leave it as it is - without default values

Reply:
I agree.

Reply:
So can we use `"js"` as the default value.